### PR TITLE
Autorename

### PR DESCRIPTION
**Pull request by taniwha**
This allows KIS items to specify they want to use Vessel.AutoRename based on the originating vessel's name rather than the item part's title. The default is to use the item part's title, but setting vesselAutoRename to true in the MODULE block alters this. This is most useful for EL's survey stakes, but other mods may find it useful.

### CODE REVIEW

File: Source/KIS_Shared.cs [hunk 1]
```diff
@@ -1363,7 +1363,13 @@ public enum MessageAction {
     Vessel newVessel = newPart.gameObject.AddComponent<Vessel>();
     newVessel.id = Guid.NewGuid();
     if (newVessel.Initialize(false)) {
-      newVessel.vesselName = newPart.partInfo.title;
+      var item = newPart.FindModuleImplementing <ModuleKISItem> ();
+      if (item != null && !item.vesselAutoRename) {
```
Comment (ihsoft) [inline Source/KIS_Shared.cs]: This condition doesn't keep the default behavior. In case of there is no module `ModuleKISItem`, then new naming mechanism will be used.

File: Source/KIS_Shared.cs [hunk 2]
```diff
@@ -1354,7 +1354,7 @@ public enum MessageAction {
 
   /// <summary>Creates a new vessel from a single part.</summary>
   /// <remarks>Initially the part must belong to some vessel.</remarks>
-  static IEnumerator WaitAndMakeLonePart(Part newPart, OnPartReady onPartReady) {
+  static IEnumerator WaitAndMakeLonePart(Part newPart, Part fromPart, OnPartReady onPartReady) {
```
Comment (ihsoft) [inline Source/KIS_Shared.cs]: You don't need an extra parameter. Initially, `newPart` belongs to the from vessel. Simply save the vessel (or the name) at the beginning of the method.

**Review by ihsoft — CHANGES_REQUESTED** [review]
See comments. Also, I see a lot of other unrelated changes. Is it your working version?

File: Source/KISAddonConfig.cs [hunk 3]
```diff
@@ -22,52 +23,133 @@ sealed class KISAddonConfig : MonoBehaviour {
...
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
+          AddPodInventories (avPart.partPrefab, avPart.partPrefab.CrewCapacity);
+        }
+        yield return null;
```
Comment (ihsoft) [inline Source/KISAddonConfig.cs]: Returning `null` will result in loading one part per a frame. On the low-end machines with many addons it may take forever to complete. This is definitely not the case when smooth loading is better than quick loading.  At least do some batching here. And, in general, let's extract this refactoring into a separate PR. I'd like to test it before saying "yes".

File: Source/KISAddonConfig.cs [hunk 4]
```diff
@@ -22,52 +23,133 @@ sealed class KISAddonConfig : MonoBehaviour {
...
+      loadedPartCount = PartLoader.LoadedPartsList.Count;
+      StartCoroutine (LoadInventories ());
+    }
+  }
+
+  public void Awake() {
+    List<LoadingSystem> list = LoadingScreen.Instance.loaders;
+    if (list != null) {
+      for (int i = 0; i < list.Count; i++) {
+        if (list[i] is KISConfigLoader) {
```
Comment (ihsoft) [inline Source/KISAddonConfig.cs:128]: Loading the configs is a cheap operation. Just load it in KISPodInventoryLoader.

**Comment by taniwha** [comment]
It's not about expense, but allowing other mods to call AddPodInventories in their OnLoad methods. ie, configs need to be loaded before parts so part modules can (indirectly) reference them.

I'll add some comments.

**Comment by taniwha** [comment]
Most of the other changes (makefile, warnings) were just to get a build (using make and -warnaserror). The inventory set would up on the same PR due to not being sure what to do with it.

File: Source/KIS_Shared.cs (hunk 1 above)
Comment (taniwha) [reply Source/KIS_Shared.cs]: oops, inverted logic.

File: Source/KISAddonConfig.cs (hunk 4 above)
Comment (taniwha) [reply Source/KISAddonConfig.cs:128]: See overall reply, but I realized I did add comments.

File: Source/KIS_Shared.cs (hunk 2 above)
Comment (taniwha) [reply Source/KIS_Shared.cs]: Ah, ok. I just tested and that does indeed work nicely. Have a problem with ground attaching, though (probably something went wrong in the config patch).

Comment (taniwha) [reply Source/KIS_Shared.cs]: Turns out the attach problem was PEBCAK: I was using the drill instead of EL's mallet.

**Comment by ihsoft** [comment]
Can we split the PR into two (or more) PRs? I'm OK with the auto rename thing. However, the mod loading refactoring is not that obvious to me at this point. It would be better to discuss/review it separately.

**Comment by taniwha** [comment]
Sure, I'll sort it out in the next day or two.

**Comment by ihsoft** [comment]
Thanks.

Btw, take a look at the [coding style definitions](https://github.com/ihsoft/KIS/blob/master/Source/README.md). I'm not a "grammar-nazi", but given how much time I spend working with the code, I'd prefer it to be following some syntax rules :)

**Comment by taniwha** [comment]
Closing in favor of newer PRs.
